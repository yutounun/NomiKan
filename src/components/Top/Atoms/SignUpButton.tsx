import React from "react";
import HowToRegOutlinedIcon from "@mui/icons-material/HowToRegOutlined";
import MyButton from "components/Commons/Atoms/MyButton";

function SignUpButton() {
  return (
    <MyButton
      startIcon={<HowToRegOutlinedIcon />}
      variant="outlined"
      value="会員登録"
      color="primary"
      size="large"
    />
  );
}

export default SignUpButton;
