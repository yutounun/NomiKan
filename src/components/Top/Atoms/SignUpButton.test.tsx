import { render, screen } from "@testing-library/react";
import React from "react";
import SignUpButton from "./SignUpButton";

describe("SignUpButton", () => {
  it("renders correctly", () => {
    render(<SignUpButton />);
    expect(screen.getByTestId("button")).toBeInTheDocument();
    expect(screen.getByText("会員登録")).toBeInTheDocument();
  });
  it("takes a snapshot", () => {
    const { asFragment } = render(<SignUpButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
