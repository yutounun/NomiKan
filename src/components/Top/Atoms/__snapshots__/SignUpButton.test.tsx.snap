// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpButton takes a snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
    data-testid="button"
    style="width: 200px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="HowToRegOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zM5 18c.2-.63 2.57-1.68 4.96-1.94l2.04-2c-.39-.04-.68-.06-1-.06-2.67 0-8 1.34-8 4v2h9l-2-2H5zm15.6-5.5-5.13 5.17-2.07-2.08L12 17l3.47 3.5L22 13.91z"
        />
      </svg>
    </span>
    会員登録
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
